### ISSUE TITLE
$_FILES empty

### ISSUE THREAD

**Issue by mageho** [SchumacherFM/magento1-pictureperfect#2]
Good morning,

I need a hand here, I do not find any help on a particular problem we encounter.

We tried to disable external modules that affect the admin products grid without success
I think it related to our php / apache hosting but without a clue.

shell_exec is enabled
the directory (var/pictureperfect) was not created automatically (created by FTP, 777 CHMOD permission).

In the magento log, I have this JSON return : 

```
2016-10-04T08:17:14+00:00 DEBUG (7): Array
(
    [0] => _getUploadedFilesNames
    [1] => Array
        (
        )

    [2] => Array
        (
            [rand] => y4b8wf8s0nxldte29
        )

    [3] => Array
        (
            [form_key] => EWsQ9L9RJX8F22fs
            [productId] => 7327
            [file] => {"name":"2.jpg","extra":{"nameNoExtension":"2","extension":"jpg","fileID":0,"uniqueID":0,"groupID":0,"prettySize":"1.52 kb","started":"2016-10-04T08:12:27.908Z","ended":"2016-10-04T08:12:27.910Z"}}
            [bdReqId] => 1
            [bdReqCount] => 1
            [bdTotalFiles] => 1
            [binaryData] => Array
                (
                    [1] => [object Object]
                )

        )

)
```

It seems the GLOBAL variable $_FILES is empty.
We tried several web browser also.

Could you advice us about this problem, apparently due at our hosting environment (we are on a dedicated hosting) ?

Thank you

**Comment by SchumacherFM** [comment]
Hi! Thank you for using PicturePerfect. First ever feedback :-)

I cannot really help you  but please have a look here http://stackoverflow.com/questions/3586919/why-would-files-be-empty-when-uploading-files-to-php or google for `magento $_FILES is empty`.

**Comment by mageho** [comment]
Superbe, thank you for your fast answer.

Special thanks to your module, which is very handy for a large catalog.

I tried before post to your github repositorie.

The difficulty I have, Is I am not at ease with JavaScript, Especially with custom code.
All the posts I found, on AJAX & PHP image upload, was written with the support of jQuery.

With your custom XHR, I didn't find a clue.

Your module works for you on Magento 1.9.2.4 ?

I will try to investigate further.

Cheers from France,

Have a nice day

**Comment by SchumacherFM** [comment]
I have never tested it with M 1.9.*

If you can find the bug and fix it, send me a pull request and I'll merge it!

**Comment by mageho** [comment]
Hard time.

We solved our problem. Caused by PicturePerfect.prototype._getBlob function
line 590
file js/pictureperfect/adminhtml/pictureperfect.js

To begin with, we debug the code with console.log.
We identify a portion of code was missing during fileSlice.call

the content-type.

To debug to begin with 👍

```
console.log('blobFish - NOT WORKING', blobSlicedContainer.eachSlice(maxFilesPerReq));
console.log('blobFish - WORKING', new Array(new Array(newBlob)));
```

Console JS Output :

> blobFish - NOT WORKING 
> [Blob] 
>   0: Blob
>     size: 62730
>     **type: ""**

> blobFish - WORKING 
> Array[1]
>   0: Array[1]
>      0: Blob
>         size: 62730
>          type: "image/jpg

We modified the line 615
`blobSlicedContainer.push(self._fileSlice.call(newBlob, fsStart, fsEnd));`
to
`blobSlicedContainer.push(self._fileSlice.call(newBlob, fsStart, fsEnd, 'image/jpg'));`

Also, the actual variable blobFish.blobber deliver no result, we needed it to nested in a array 
Line 628
`blobFish.blobber = blobSlicedContainer.eachSlice(maxFilesPerReq);`
to
`blobFish.blobber = new Array(blobSlicedContainer.eachSlice(maxFilesPerReq));`

Google Chrome Version Version 53.0.2785.143 (64-bit)
Magento 1.9.2.4

We tried with one and several files with success.
No anomalies.

Do not understand very well our change, no familiar with javascript and coding in general.
Just with logic.

Also, why the image/jpg declaration, it doesn't work with other type ?

So far so good for us, 

I take my time to explain our solution,

Hope it will help you.

Cheers
